lib/coverband/collectors/coverage.rb

```diff
@@ -89,12 +89,17 @@ def initialize
         raise NotImplementedError, 'Coverage needs Ruby > 2.3.0' if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.3.0')
 
         require 'coverage'
-        if Coverage.ruby_version_greater_than_or_equal_to?('2.6.0')
-          ::Coverage.start(oneshot_lines: Coverband.configuration.use_oneshot_lines_coverage) unless ::Coverage.running?
-        elsif Coverage.ruby_version_greater_than_or_equal_to?('2.5.0')
-          ::Coverage.start unless ::Coverage.running?
+        if defined?(SimpleCov) && defined?(Rails) && defined?(Rails.env) && Rails.env.test?
+          puts "Coverband: detected SimpleCov in test Env, allowing it to start Coverage"
```

Comment:
What happens if `Coverage.start` is called twice? I had thought it raises an error private to 2.5 but i just tried in 2.4.4 and nothing seemed to happen.

```
➜ ruby -v
ruby 2.4.4p296 (2018-03-28 revision 63013) [x86_64-darwin17]
➜ irb
irb(main):001:0> require 'coverage'
=> true
irb(main):002:0> Coverage.start
=> nil
irb(main):003:0> Coverage.start
=> nil
irb(main):004:0>
```

Reply:
So it depends what is going on and the version of Ruby, but it raises an error for sure if you call start with different options. We call `Coverage.start` and now SimpleCov calls `Coverage.start(lines: true, branches: true)` so you get this error, when SimpleCov tries to start up.

```cannot change the measuring target during coverage measurement
/Users/danmayer/.rvm/gems/ruby-2.6.2/gems/simplecov-0.18.5/lib/simplecov.rb:354:in `start'
/Users/danmayer/.rvm/gems/ruby-2.6.2/gems/simplecov-0.18.5/lib/simplecov.rb:354:in `start_coverage_with_criteria'
/Users/danmayer/.rvm/gems/ruby-2.6.2/gems/simplecov-0.18.5/lib/simplecov.rb:343:in `start_coverage_measurement'
/Users/danmayer/.rvm/gems/ruby-2.6.2/gems/simplecov-0.18.5/lib/simplecov.rb:58:in `start'
/Users/danmayer/projects/coverband_demo/test/test_helper.rb:3:in `<top (required)>'
```

Reply:
Right. Just wondering if we could eventually simplify this part of the code by just catching the exception that gets thrown if you call start in certain ruby versions. I usually don't like to have logic that depends on exceptions being thrown but this could reduce the number of branches we have here.

Reply:
It would be hard since it is thrown from a simplecov call, and all their call stack... I guess another option is to detect SimpleCov and the version and start coverage with the same arguments that they do... in some sense I don't want to get to tied to them, and think the warning might be less complexity for now

Reply:
Gotcha. This does sound challenging.